sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PropertiesAccessCorrectField.cs

```diff
@@ -135,14 +182,14 @@ protected override bool ShouldIgnoreAccessor(IMethodSymbol accessorMethod)
             if (strippedExpression is IdentifierNameSyntax &&
                 semanticModel.GetSymbolInfo(strippedExpression).Symbol is IFieldSymbol field)
             {
-                return new FieldData(accessorKind, field, strippedExpression);
+                return new FieldData(accessorKind, field, strippedExpression, useFieldLocation);
             }
             // Check for "this.foo"
             else if (strippedExpression is MemberAccessExpressionSyntax member &&
```

Comment:
The else is not really needed since the first if returns directly.

Reply:
I prefer to keep it for readability. It is else-if logic.